aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -199,6 +199,11 @@ static void addmm_impl_cpu_(
   auto m2_strides = m2.strides();
   auto m2_sizes = m2.sizes();
 
+  TORCH_CHECK(self.device() == kCPU && m1.device() == kCPU &&
```

Comment:
indeed, multidispatch will always send to cuda if one of the tensors is cuda, so this is not needed